src/components/SelectButtonGroup/SelectButtonGroup/SelectButtonGroup.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+
+interface Props {
+    /** SelectButton components */
+    children: NotEmptyReactNode;
+    /**
+     * A function to be called if a button is pressed.
+     * This prop, if defined will overwrite the onSelect on children
+     */
+    onSelect?: (value: string) => void;
```

Comment:
If it is `isMultiselect` i think better to pass all selected values.
Not clear if we uncheck value so what `onSelect` should return?
Passed all selected values solve this problem.

Reply:
do you mean: `onSelect?: (value: string | number | boolean) => void | (selection: [name: string, value: string | number | boolean]) => void` where the first typing is for single select, and the second typing is for multiselect?

Reply:
The `name` for selection will be the same, correct?
Originally the idea was implemented  kind of `select/option` that supports `multiply`
https://mdbootstrap.com/docs/jquery/forms/multiselect/
So maybe better:
```
onChange?: ({ name: string, values:  [] }) => void
```

Reply:
For `radio` it will be the mostly the same
```
onChange?: ({ name: string, value:  string }) => void
```

Reply:
but I think maybe better pass arguments, not object:
```
 onChange?: (value: [string] |  string, name: string) => void
```
Then in external callback function we can use any variable names:
```
handleChange(selectedValues) {

}

<SelectButtonGroup
...
onChange={handleChange}
/>

Reply:
`name` is the same for all of them if they are radio buttons but not if they are checkboxes. Hence my suggestion to send it back in a similar format to the native elements. Your example relates to a select box, not a group of checkboxes. 

Native behaviour for radio is: 
`name=value` where `name` is the same for all radio buttons in the group and `value` is unique for each individual button.
So here I suggest to return `onSelect: (value: string) => void`

Native behaviour for checkbox is: 
`name=value` where `name` is unique to every checkbox on the page and `value=true` if not otherwise defined. This will be returned for individual checked checkbox on the form. 
So in multiselect mode I suggest `onSelect: (selection: [name: string, value: string][]) => void`. Alternatively, to be consistent with the radio buttons we can use `onSelect: (selection: string[]) => void` which will return an array of the values of the selected options. So name is not a required prop.

Reply:
To sum it up, as suggested the implementation will be based on select. So the callback can have the following signature: `onChange: (selection: string | string[]) => void`. Do you agree?

Reply:
So maybe let's add the `name` to  `SelectButtonsGroup` as well.
Take a look at the case where it will useful:
```
handleChange(value, name) {
  data[name] = value;
}

<SelectButtonsGroup name="nameOne" onChange={handleChange}>....
<SelectButtonsGroup name="nameTwo" onChange={handleChange}>....
```
In the example above we can easily distingue what exactly part of data was changed.
Otherwise, without the `name` prop we need to write a distinct handler for each `SelectButtonsGroup`:
```
<SelectButtonsGroup onChange={handleNameOneChange}>....
<SelectButtonsGroup onChange={handleNameTwoChange}>....
```